### PR TITLE
[scudo] Double frees result in chunk state error

### PR DESCRIPTION
**Pull request by cferris1000**
Fixes bug where a device that supports tagged pointers doesn't use
the tagged pointer when computing the checksum.

Add tests to verify that double frees result in chunk state error
not corrupted header errors.

### CODE REVIEW

**Comment by cferris1000** [comment]
I didn't add secondary tests because that's a bit trickier since we tend to wipe the header in many cases.

File: compiler-rt/lib/scudo/standalone/combined.h [hunk 1]
```diff
@@ -1252,22 +1252,25 @@ class Allocator {
...
-    if (LIKELY(!useMemoryTagging<AllocatorConfig>(Options))) {
+    if (LIKELY(!useMemoryTagging<AllocatorConfig>(Options)))
       Header->OriginOrWasZeroed = 0U;
-      if (BypassQuarantine && allocatorSupportsMemoryTagging<AllocatorConfig>())
-        Ptr = untagPointer(Ptr);
-      BlockBegin = getBlockBegin(Ptr, Header);
-    } else {
+    else
       Header->OriginOrWasZeroed =
           Header->ClassId && !TSDRegistry.getDisableMemInit();
```
Comment (ChiaHungDuan) [inline compiler-rt/lib/scudo/standalone/combined.h:1259]: Small nit:

It's a multiple lines statement so I would add brackets here

**Review by ChiaHungDuan — APPROVED** [review]

File: compiler-rt/lib/scudo/standalone/combined.h (hunk 1 above)
Comment (cferris1000) [reply compiler-rt/lib/scudo/standalone/combined.h]: Done.

**Review by ChiaHungDuan — APPROVED** [review]

**Comment by cferris1000** [comment]
Originally, I thought that the test might fail with a real crash without an abort when mte enabled. But I verified that it does properly detect the chunk state even when MTE is enabled.

**Pull request merged**